### PR TITLE
Handle reset by default in Form

### PR DESCRIPTION
**Pull request by nbouscal**
This will make `<button type="reset">` inside the form trigger the form's reset handler. This seems like expected behavior for the `Form` component, because it's almost the exact same functionality as the `submit` case.

I'm not sure which documentation changes you'll want to make if you accept this. I've made the one that seems most clearcut.

### CODE REVIEW

File: docs/api/form.md [hunk 1]
```diff
@@ -4,12 +4,12 @@ title: <Form />
 custom_edit_url: https://github.com/jaredpalmer/formik/edit/master/docs/api/form.md
 ---
 
-Form is a small wrapper around an HTML `<form>` element that automatically hooks into Formik's `handleSubmit`. All other props are passed directly through to the DOM node.
+Form is a small wrapper around an HTML `<form>` element that automatically hooks into Formik's `handleReset` and `handleSubmit`. All other props are passed directly through to the DOM node.
```
Comment (jaredpalmer) [inline docs/api/form.md]: I would reverse the order here. handleSubmit should come first and reset second?

Comment (nbouscal) [reply docs/api/form.md]: Yep that sounds good, I just went with alphabetical but agree that submit is more important/common and should go first.

Comment (jaredpalmer) [reply docs/api/form.md]: Question for you. How does this work with buttons without `type`? Just confirming that submit still takes precedence, yes?

Comment (nbouscal) [reply docs/api/form.md]: It should be the same default behavior, which iirc is browser-dependent. I haven’t tested it super thoroughly, but this should only affect buttons that explicitly have type reset. (No browsers default to reset, only submit or button.)

**Comment by jaredpalmer** [comment]
@eonwhite Are we good with this?

**Comment by jaredpalmer** [comment]
Will cut a release.

**Comment by jaredpalmer** [comment]
Tomorrow.

**Pull request merged**